internal/encryption/keyprovider/openbao/config.go

```diff
@@ -0,0 +1,97 @@
+package openbao
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	openbao "github.com/openbao/openbao/api"
+	"github.com/opentofu/opentofu/internal/encryption/keyprovider"
+)
+
+type Config struct {
+	Address string `hcl:"address,optional"`
+	Token   string `hcl:"token,optional"`
+
+	KeyName   string `hcl:"key_name"`
+	KeyLength int    `hcl:"key_length,optional"`
```

Comment:
A separate type would be helpful for such cases, agree. I don't think `Set` will be beneficial here since it's only 3 possible values: [033bddb](https://github.com/opentofu/opentofu/pull/1436/commits/033bddbbeba08f084049882aa2fab9818a654a9a)